pkg/logql/log/distinct.go

```diff
@@ -0,0 +1,37 @@
+package log
+
+type distinctFilter struct {
+	datas  map[string]map[string]int
```

Comment:
How do you keep the state across multiple queriers?

Reply:
Great point, I didn't notice this detail.
Maybe we can avoid logql to be executed in multiple queriers by making the `Shardable() bool` function of `DistinctFilterExpr` struct return false, just like calculating the 99th quantile.

There are too many details to consider when introducing a state
![image](https://user-images.githubusercontent.com/9583245/232650358-377c7038-8623-48a5-a7e9-cd0a9db9e4cf.png)

Reply:
Great idea. Could add it?

Reply:
You always did 😊